packages/twenty-server/src/modules/mail-sender/workflow-actions/send-email.workflow-action.ts

```diff
@@ -73,37 +73,35 @@ export class SendEmailWorkflowAction {
 
   async execute({
     step,
-    payload,
+    context,
   }: {
     step: WorkflowSendEmailStep;
-    payload: {
-      email: string;
-      [key: string]: string;
-    };
+    context: Record<string, any>;
   }): Promise<WorkflowActionResult> {
     const emailProvider = await this.getEmailClient(step);
+    const { email, body, subject } = step.settings.input;
 
     try {
       const emailSchema = z.string().trim().email('Invalid email');
 
-      const result = emailSchema.safeParse(payload.email);
+      const result = emailSchema.safeParse(email);
```

Comment:
You need to safeParse the inferredEmail, not the email from input as it will fail if you use a variable